lib/sqlite3/database.rb

```diff
@@ -691,6 +691,22 @@ def readonly?
       @readonly
     end
 
+    # Sets a #busy_handler that releases the GVL between retries,
+    # but only retries up to the indicated number of +milliseconds+.
+    # This is an alternative to #busy_timeout, which holds the GVL
+    # while SQLite sleeps and retries.
+    def busy_handler_timeout=( milliseconds )
+      timeout_seconds = milliseconds.fdiv(1000)
+      retry_interval = 0.001 # 1 millisecond
```

Comment:
You can do without that local variable, that will be one less indirection in the proc.